app/src/main/java/com/example/parkingsystem/util/ParkingLotReservationValidated.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.parkingsystem.util
+
+import com.example.parkingsystem.database.ParkingReservationDatabase
+import com.example.parkingsystem.entity.ParkingLotReservation
+
+class ParkingLotReservationValidated(var database: ParkingReservationDatabase) {
+
+    fun canBeReserved(reservation: ParkingLotReservation): Boolean {
+        val parkingLot = reservation.parkingLot
+        if (database.hashMapReservation.containsKey(parkingLot)) {
+            val reservations: MutableList<ParkingLotReservation>? = database.hashMapReservation[parkingLot]
```

Comment:
same here avoid nullable variable

Reply:
I can´t change this one, because it doesn't work (there's no error but it doesn't compare anymore, so there's no overlap validation either)